flink/v1.16/flink/src/main/java/org/apache/iceberg/flink/source/FlinkSplitPlanner.java

```diff
@@ -86,11 +87,23 @@ static CloseableIterable<CombinedScanTask> planTasks(
       IncrementalAppendScan scan = table.newIncrementalAppendScan();
       scan = refineScanWithBaseConfigs(scan, context, workerPool);
 
+      if (context.startTag() != null && table.snapshot(context.startTag()) != null) {
```

Comment:
we should add Preconditions check for `table.snapshot(context.startTag())`. Invalid reference should result in an exception. this applies to other refs.